File: 3014-minimum-number-of-pushes-to-type-word-i/3014-minimum-number-of-pushes-to-type-word-i.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        freq = [0] * 26
        for ch in word:
            freq[ord(ch) - 97] += 1
        freq.sort(reverse=True)

        res = 0
        for i, cnt in enumerate(freq):
            if cnt == 0:
                break
            if i < 8:
                res += cnt
            elif i < 16:
                res += cnt * 2
            elif i < 24:
                res += cnt * 3
            else:
                res += cnt * 4
        return res

File: 3014-minimum-number-of-pushes-to-type-word-i/3014-minimum-number-of-pushes-to-type-word-i.cpp
class Solution {
public:
    int minimumPushes(string word) {
        int n = word.length(), res = 0;
        vector<int> v(26, 0);

        for (char c : word) v[c - 'a']++;
        
        sort(v.rbegin(), v.rend());

        for (int i = 0; i < size(v); i++) {
            if (v[i] == 0) break;

            if (i < 8) res += v[i];

            else if (i < 16) res += v[i] * 2;

            else if (i < 24) res += v[i] * 3;

            else res += v[i] * 4;
        }
        return res;
    }
};